spring/src/main/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerFactoryProcessor.java

```diff
@@ -121,6 +61,75 @@ public int getOrder() {
   @Override
   public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry)
       throws BeansException {
-    this.beanDefinitionRegistry = registry;
+    if (registry instanceof BeanFactory) {
+      for (String name : registry.getBeanDefinitionNames()) {
+        Class<?> cls = ((BeanFactory) registry).getType(name);
```

Comment:
seems all beans are inspected for @ KubernetesInformers , which IMO is suboptimal. maybe some restriction (as prev existed) is not a bad idea. How about KubernetesInformers at Reconcilers only?

Isn't it problematic that there's no check for duplication and two Components might register same PodInformer twice. Cause now you have different sources of KubernetesInformers, while prev it was only through singleton SharedInformerFactory.

Reply:
I think that’s beyond the scope of this PR. The inefficiency is also insignificant - either we have to check all beans or Spring does. The work gets done one way or the other. (Unless we really want to restrict the type of the annotated beans, which was the original design, but doesn’t make any sense to me.)

Reply:
> Isn't it problematic that there's no check for duplication and two Components might register same PodInformer twice. Cause now you have different sources of KubernetesInformers, while prev it was only through singleton SharedInformerFactory.

How about at least to create a test with two sources of KubernetesInformers to see (and be prepared) what will happen?

Reply:
Not sure what you mean by "2 sources". There is only one `SharedIndexInformer` per `ApiType` and it's the same one as the code in master, so where would the second one come from? Why would anyone need 2?

Reply:
Till now one was supposed to create **one** `ShatedInformerFactory` annotated with `KubernetesInformers`. This is what I call one-single-source of `KubernetesInformers`.
With this review one might provide more components annotated with KubernetesInformers. (this is what I call multiple sources of ...) such as:

```
    @KubernetesInformers({
      @KubernetesInformer(
          apiTypeClass = V1Pod.class,
          apiListTypeClass = V1PodList.class,
          groupVersionResource = @GroupVersionResource(resourcePlural = "pods"))
    })
    @Component
    static class SourceOneOfKubernetesInformers {}

    @KubernetesInformers({
      @KubernetesInformer(
          apiTypeClass = V1Pod.class,
          apiListTypeClass = V1PodList.class,
          groupVersionResource = @GroupVersionResource(resourcePlural = "pods"))
    })
    @Component
    static class SourceTwoOfKubernetesInformers {}
```
So I'm concerned about such a scenario, and how it's handled by the code.

Reply:
There was nothing stopping you from doing that before (but on the same bean). It would fail at runtime, so you would know it was wrong, even if the exception might be confusing without some polish. I don't see any reason to fix it as part of this PR because it's not a new problem and it's going to fail at runtime anyway.

Reply:
Agree with you. 
Still spotting the duplication on a single ShatedInformerFactory seems much easier that spotting it on two different components (in two diff packages) if the message is not hinting you. Seems as good-to-have, and not must-to-have for this PR.